packages/core/chrome/core-chrome-browser-internal/src/ui/header/screen_reader_a11y.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { FC, useState, useEffect } from 'react';
+import useObservable from 'react-use/lib/useObservable';
+import { EuiScreenReaderLive } from '@elastic/eui';
+
+import type { HeaderProps } from './header';
+
+const DEFAULT_TITLE = 'Elastic'; // This may need to be DRYed out with https://github.com/elastic/kibana/blob/main/packages/core/rendering/core-rendering-server-internal/src/views/template.tsx#L34
+const SEPARATOR = ' - ';
+
+export const ScreenReaderRouteAnnouncements: FC<{
+  breadcrumbs$: HeaderProps['breadcrumbs$'];
+}> = ({ breadcrumbs$ }) => {
+  const [routeTitle, setRouteTitle] = useState('');
+  const fallbackBreadcrumbs = useObservable(breadcrumbs$, []);
+
+  useEffect(() => {
+    if (fallbackBreadcrumbs.length) {
+      const breadcrumbText: string[] = [];
+
+      // Reverse the breadcrumb title order and ensure we only pick up valid strings
+      fallbackBreadcrumbs.reverse().forEach((breadcrumb) => {
+        if (typeof breadcrumb.text === 'string') breadcrumbText.push(breadcrumb.text);
+      });
+      breadcrumbText.push(DEFAULT_TITLE);
+
+      setRouteTitle(breadcrumbText.join(SEPARATOR));
+    } else {
+      // Don't announce anything during loading states
+      setRouteTitle('');
+    }
+  }, [fallbackBreadcrumbs]);
```

Comment:
(see above comment) code suggestion w/ the `fallbackBreadcrumbs`->`breadcrumbs` change if that helps at all
```suggestion
    if (breadcrumbs.length) {
      const breadcrumbText: string[] = [];

      // Reverse the breadcrumb title order and ensure we only pick up valid strings
      breadcrumbs.reverse().forEach((breadcrumb) => {
        if (typeof breadcrumb.text === 'string') breadcrumbText.push(breadcrumb.text);
      });
      breadcrumbText.push(DEFAULT_TITLE);

      setRouteTitle(breadcrumbText.join(SEPARATOR));
    } else {
      // Don't announce anything during loading states
      setRouteTitle('');
    }
  }, [breadcrumbs]);
```

Reply:
Nice!!! [0cb69ae](https://github.com/elastic/kibana/pull/150461/commits/0cb69ae4529e80663c413427e74ddcd2f7d9e1fa) for changes!